openid-4-verifiable-presentations-1_0.md

```diff
@@ -2348,19 +2182,33 @@ The security properties that are normally provided by the Client Identifier are
 
 The audience for the response (for example, the `aud` value in a Key Binding JWT) MUST be the Origin, prefixed with `origin:`, for example `origin:https://verifier.example.com/`. This is the case even for signed requests. Therefore, when using OpenID4VP over the DC API, the Client Identifier is not used as the audience for the response.
 
-# Credential Format Specific Parameters {#format_specific_parameters}
+# Credential Format Specific Parameters and Rules {#format_specific_parameters}
 
-OpenID for Verifiable Presentations is Credential Format agnostic, i.e., it is designed to allow applications to request and receive Verifiable Presentations and Verifiable Credentials in any Credential Format. This section defines a set of Credential Format specific parameters for some of the known Credential Formats. For the Credential Formats that are not mentioned in this specification, other specifications or deployments can define their own set of Credential Format specific parameters.
+OpenID for Verifiable Presentations is Credential Format agnostic, i.e., it is designed to allow applications to request and receive Verifiable Presentations and Verifiable Credentials in any Credential Format. This section defines a set of Credential Format specific parameters and rules for some of the known Credential Formats. For the Credential Formats that are not mentioned in this specification, other specifications or deployments can define their own set of Credential Format specific parameters.
 
 ## W3C Verifiable Credentials
 
-W3C Verifiable Credentials may use an additional parameter for the `descriptor_map` with the `presentation_submission`: The `path_nested` object inside an Input Descriptor Mapping Object is used to describe how to find a returned Credential within a Verifiable Presentation, and contains a `format` parameter with the Credential format identifier as a value and a `path` parameter with a relative path to the Verifiable Credential. Non-normative examples can be found further in this section.
+todo: define where the claims path operates
+todo: is proof_type_value correct?
+
+### DCQL
+
+#### Parameters in the `meta` parameter in Credential Query
+
+The following is a W3C Verifiable Credentials specific parameter in the meta parameter in a Credential Query as defined in (#credential_query):
+
```

Comment:
I think a custom `type_values` which is an array (or array of arrays to allow multiple type combinations) would be good to add here. My reasoning is that there is `doctype_values` and `vct_values`, so this will ensure consistency. Also, the `doctype_values` and `vct_values` are required, so also adding `type_values` means we also have a required meta type filter for w3c credentials (which was an issue with PEX)

Reply:
Yeah, fehlt the same way to me for the W3C Parts: https://github.com/openid/OpenID4VP/pull/479#discussion_r2019986109. I do believe it would be a good idea to pull json-ld types matching out into format specific parts.

Reply:
The point about making it mandatory is a good one, as well as the lack of consistency at the moment. I'll change it to type_values.

Reply:
That will change with this PR: #480